### PR TITLE
tsp, json-merge-patch, fix issue on patch for discriminator property

### PR DESCRIPTION
**Pull request by haolingdong-msft**
Fix https://github.com/Azure/autorest.java/issues/2615

### CODE REVIEW

**Review by weidongxu-microsoft — APPROVED** [review]
maybe bump the typespec-java version as well

File: javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java [hunk 1]
```diff
@@ -821,6 +821,9 @@ private void addModelConstructor(ClientModel model, JavaVisibility constructorVi
                 if (modelDefinesProperty(model, polymorphicProperty)) {
                     constructor.line("this." + polymorphicProperty.getName() + " = " + discriminatorValue + ";");
                 }
+                if (ClientModelUtil.isJsonMergePatchModel(model)) {
+                    constructor.line(String.format("this.updatedProperties.add(\"%s\");", polymorphicProperty.getName()));
```
Comment (srnagar) [inline javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java:825]: nit: avoid using `String.format`. It's performance is very bad. We can just use string concatenation.

```suggestion
                    constructor.line("this.updatedProperties.add(\"" + polymorphicProperty.getName() + "\");");
```

**Review by srnagar — APPROVED** [review]

File: typespec-extension/package.json
Comment (weidongxu-microsoft) [inline typespec-extension/package.json:1]: run install and update lock file as well.

**Comment by weidongxu-microsoft** [comment]
sorry, forget the changelog. need to update that too.

File: javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java (hunk 1 above)
Comment (haolingdong-msft) [reply javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java]: Fixed for the two prs on json merge patch .

**Pull request merged**